tutoraspects/templates/aspects/apps/superset/pythonpath/openedx_sso_security_manager.py

```diff
@@ -88,22 +101,73 @@ def access_token(self):
         """
         return self.get_oauth_token().get("access_token")
 
-    def _get_user_roles(self, username):
+    def _create_clickhouse_user(self, clickhouse_username, user_roles):
+        """
+        Create a clickhouse user with appropriate access.
+        """
+        # FIXME: Should probably make some kind of password hashed off of SECRET_KEY
+        #   that we can re-hash for use in the DB_CONNECTION_MUTATOR
+        make_clickhouse_query(f"CREATE USER IF NOT EXISTS '{clickhouse_username}' "
+                              f"IDENTIFIED BY 'password';")
+        make_clickhouse_query(
+            f"GRANT CREATE TEMPORARY TABLE ON *.* TO '{clickhouse_username}';"
+        )
+
+        self._grant_or_revoke_clickhouse_role(
+            clickhouse_username, "openedx_instructor", "openedx" in user_roles
+        )
+        self._grant_or_revoke_clickhouse_role(
+            clickhouse_username, "openedx_superuser", "admin" in user_roles
+        )
+        self._grant_or_revoke_clickhouse_role(
+            clickhouse_username, "openedx_administrator", "alpha" in user_roles
+        )
+
+        make_clickhouse_query(f"SET DEFAULT ROLE ALL TO '{clickhouse_username}';")
```

Comment:
This makes all of the above roles active for the user by default, necessary for them to be useful here.